### PR TITLE
change color constructor and FbxLoader return value

### PR DESCRIPTION
**Pull request by Devilsparta**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/three/three-fbxloader.d.ts [hunk 1]
```diff
@@ -1,14 +1,17 @@
 // https://github.com/mrdoob/three.js/blob/master/examples/js/loaders/FBXLoader.js
 
 import { LoadingManager } from "./three-core";
-import {Group} from "./three-core";
+import { AnimationClip, Group } from "./three-core";
 
+interface ILoadedModel extends Group {
+	animations?: AnimationClip[];
```
Comment (efokschaner) [inline types/three/three-fbxloader.d.ts]: As far as I can tell animations will never be undefined.

File: types/three/three-fbxloader.d.ts [hunk 2]
```diff
@@ -1,14 +1,17 @@
 // https://github.com/mrdoob/three.js/blob/master/examples/js/loaders/FBXLoader.js
 
 import { LoadingManager } from "./three-core";
-import {Group} from "./three-core";
+import { AnimationClip, Group } from "./three-core";
 
+interface ILoadedModel extends Group {
```
Comment (efokschaner) [inline types/three/three-fbxloader.d.ts]: The implementation seems to refer to its return value as a `sceneGraph` so maybe something like `FbxSceneGraph` would be a more appropriate name? Up to you though.

File: types/three/three-core.d.ts [hunk 3]
```diff
@@ -3278,9 +3278,7 @@ export interface HSL {
  * @see <a href="https://github.com/mrdoob/three.js/blob/master/src/math/Color.js">src/math/Color.js</a>
  */
 export class Color {
-    constructor(color?: Color);
-    constructor(color?: string);
-    constructor(color?: number);
+    constructor(color?: Color|string|number);
```
Comment (efokschaner) [inline types/three/three-core.d.ts]: Just curious I assume we prefer union to overload to make it possible to actually pass a union?

**Review by efokschaner — CHANGES_REQUESTED** [review]
Thanks for the improvements, I had some suggestions.

File: types/three/three-core.d.ts (hunk 3 above)
Comment (Devilsparta) [reply types/three/three-core.d.ts]: I'm working a gaming project, and I'm going to work with many config file.Many config file uses either string or number to declare a color.So I need a function like this:

```javascript
function createColor(colorParam: string|number){
    return new Color(color);
}
```

this function cannot work.And I don't want this function like
```javascript
function createColor(colorParam:string|number){
    if(typeof colorParam === "string"){
        return new Color(color);
    }else if(typeof colorParam === "number"){
        return new Color(color)
    }
}
```

File: types/three/three-fbxloader.d.ts (hunk 2 above)
Comment (Devilsparta) [reply types/three/three-fbxloader.d.ts]: FbxSceneGraph is much better

**Review by efokschaner — APPROVED** [review]

File: types/three/three-fbxloader.d.ts [hunk 4]
```diff
@@ -1,14 +1,17 @@
 // https://github.com/mrdoob/three.js/blob/master/examples/js/loaders/FBXLoader.js
 
 import { LoadingManager } from "./three-core";
-import {Group} from "./three-core";
+import { AnimationClip, Group } from "./three-core";
 
+interface IFbxSceneGraph extends Group {
+	animations: AnimationClip[];
```
[the remark this answers is not in the document]
Comment (Devilsparta) [reply types/three/three-fbxloader.d.ts]: sorry for indentation issue. fixed :)

**Pull request merged**